dmutils/logging.py

```diff
@@ -98,11 +108,10 @@ class CustomLogFormatter(logging.Formatter):
 
     def add_fields(self, record):
         for field in self.FORMAT_STRING_FIELDS_PATTERN.findall(self._fmt):
-            record.__dict__[field] = record.__dict__.get(field)
-        return record
+            record.__dict__[field] = record.__dict__.get(field) or "-"
```

Comment:
I this is maybe obvious but this going to catch `False` and `0` and `0.00` as well as `None` now, the zeroes in particular it might be a bit weird to convert to `-` if they were a real value or something.

Reply:
ah balls - zeros.

Reply:
See, I hummed and hahed over which of these to use for about a minute but was thinking mostly about wanting to catch explicitly-set `None`s. Forget that.